sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/MetricStorage.java

```diff
@@ -45,4 +46,15 @@ MetricData collectAndReset(
       long startEpochNanos,
       long epochNanos,
       boolean suppressSynchronousCollection);
+
+  /**
+   * Determines whether this storage is an empty metric storage.
+   *
+   * <p>Uses the reference comparison since {@link EmptyMetricStorage} is singleton.
+   *
+   * @return true if is empty.
+   */
+  default boolean isEmpty() {
```

Comment:
Nit: You could make this be a static method similar to the `register` method you added, rather than adding a new instance-level interface method that's really only used to turn off storage.